rfds/116-Handling-Old-Comments/index.adoc

```diff
@@ -0,0 +1,297 @@
+----
+INSERT INTO test_schema.github_comments_view(id, status, content)
+VALUES 
+    (1, 'active', 'foo'),
+    (1, 'overwritten', 'bar'),
+    (1, 'minimized', 'foobar');
+----
+
+The `OCaml` codebase could focus only on inserting and reading from a view, the remaining machinery (and how it is supposed to work) can be kept consistent behind a view.
+
+[source,sql]
+----
+terrateam=> SELECT * FROM test_schema.github_comments_view;
+ id |   status    | content | version 
+----+-------------+---------+---------
+  1 | active      | foo     |       1
+  1 | overwritten | bar     |       2
+  1 | minimized   | foobar  |       3
+(3 rows)
+----
+
+==== Second Proposal
+
+This second proposal tries to leverage some of the SQL temporality features provider by some extensions, like https://github.com/xocolatl/periods[periods] or https://github.com/arkhipov/temporal_tables[temporal_tables]. This will work similarly to the previous approach, with the advantage that we can just leverage a single table.
+
+=== Github Implementation
+
+Github gives us two options:
+
+* The GraphQL API supports https://docs.github.com/en/graphql/reference/mutations#minimizecomment[minimizeComment] and https://docs.github.com/en/graphql/reference/mutations#unminimizecomment[unminimizeComment], which can help our noise reduction effort by minimizing outdated comments.
+* The REST API does not support these `minimize`/`unminimize` mutations, but allows for https://docs.github.com/en/rest/issues/comments?apiVersion=2022-11-28#update-an-issue-comment[updating comments], which can be used to maintain an "anchor" comment that is continuously updated with the latest status.
+
+The following example is a proof-of-concept bash script on how to update a comment via the GitHub REST API.
+
+[source,bash]
+----
+#!/usr/bin/env bash
+
+GITHUB_TOKEN=$1
+# org/repo
+REPO=$2
+# int
+PR_NUMBER=$3
+ORIGINAL_COMMENT="foo"
+UPDATED_COMMENT="<details><summary>Generation 1</summary>$ORIGINAL_COMMENT</details>\n✅ LATEST COMMENT!"
+
+response=$(curl -s -H "Authorization: token $GITHUB_TOKEN" \
+    -H "Accept: application/vnd.github+json" \
+    -X POST \
+    -d "{\"body\": \"$ORIGINAL_COMMENT\"}" \
+    "https://api.github.com/repos/$REPO/issues/$PR_NUMBER/comments")
+
+comment_id=$(echo "$response" | jq -r '.id')
+
+if [ "$comment_id" == "null" ]; then
+  echo "ERROR: $response"
+  exit 1
+fi
+
+echo "Comment ID: $comment_id"
+
+sleep 5
+
+echo "Updating comment..."
+update_response=$(curl -s -H "Authorization: token $GITHUB_TOKEN" \
+    -H "Accept: application/vnd.github+json" \
+    -X PATCH \
+    -d "{\"body\": \"$UPDATED_COMMENT\"}" \
+    "https://api.github.com/repos/$REPO/issues/comments/$comment_id")
+
+if echo "$update_response" | jq -e '.id' >/dev/null; then
+  echo "Comment updated successfully."
+else
+  echo "ERROR: $update_response"
+  exit 1
+fi
+----
+
+NOTE: Using REST limits our ability to minimize old comments, as it's clunkier than GraphQL, but totally achievable.
```

Comment:
For context: the REST API is what we use because there is a JSON schema for it so we can just automate all of the OCaml creation.  And the GraphQL tooling is not the most awesome yet in OCaml.  But, AFAIK, it's the same authentication token we use right now to use GraphQL, so we we can just write a function in  `terrat_github` that does this one GraphQL call.  We don't need any of the tooling for that, it's just a string where we need to replace some identifiers and send it via HTTP.

All that is to say, don't get hung up on "we have to use REST and we will sacrifice UX for REST" but "we use REST, but we can go outside of REST if we have to in order to solve a problem"